libraries/aws/shadow/src/aws_iot_shadow_operation.c

```diff
@@ -614,7 +603,7 @@ AwsIotShadowError_t _AwsIotShadow_CreateOperation( _shadowOperation_t ** pNewOpe
                                                    uint32_t flags,
                                                    const AwsIotShadowCallbackInfo_t * pCallbackInfo )
 {
```

Comment:
This function  `_ceateOperation` can be simplified if you follow this code structure
```
pOperation = AwsIotShadow_MallocOperation( sizeof( _shadowOperation_t ) );
if(pOperation != NULL)
{
   // Happy Path code
}
else
{
  AwsIotShadow_FreeOperation( pOperation );
}
```

It will avoid the presence of the `if( status == AWS_IOT_SHADOW_SUCCESS )` checks in the function

Reply:
We conversed offline and were able to find a more readable refactor that combines the happy-path within the single `if` block